code/go/0chain.net/blobbercore/handler/handler.go

```diff
@@ -453,12 +453,9 @@ func InsertShare(ctx context.Context, r *http.Request) (interface{}, error) {
 		ExpiryAt:                  common.ToTime(authTicket.Expiration),
 	}
 
-	existingShare, err := reference.GetShareInfo(ctx, authTicket.ClientID, authTicket.FilePathHash)
-	if err != nil {
-		return nil, err
-	}
```

Comment:
should  we check err here?  We need to know if db throw any error?

```
func GetShareInfo(ctx context.Context, clientID string, filePathHash string) (*ShareInfo, error) {
	db := datastore.GetStore().GetTransaction(ctx)
	shareInfo := &ShareInfo{}
	err := db.Table(TableName()).
		Where(&ShareInfo{
			ClientID:     clientID,
			FilePathHash: filePathHash,
		}).
		First(shareInfo).Error

	if err == gorm.ErrRecordNotFound {
		return nil, nil
	}
	if err != nil {
		return nil, err
	}
	return shareInfo, nil
}

```

Reply:
Nope, it's wrong 'tricky' implementation that causes previous bugs. I removed this check from DB level